internal/operators/ocs/validations.go

```diff
@@ -182,64 +193,69 @@ func getValidDiskCount(disks []*models.Disk) int {
 }
 
 // used to validate resource requirements for OCS excluding disk requirements and set a status message
-func (o *ocsValidator) validate(hosts []*models.Host, cpu int64, ram int64, disk int64, hostsWithDisks int64) (bool, string) {
+func (o *operator) validate(hosts []*models.Host, cpu int64, ram int64, disk int64, hostsWithDisks int64) (bool, string) {
 	var TotalCPUs int64
 	var TotalRAM int64
 	var status string
-	if int64(len(hosts)) == o.OCSRequiredHosts { // for 3 hosts
-		TotalCPUs = o.OCSRequiredCompactModeCPUCount
-		TotalRAM = o.OCSRequiredCompactModeRAMGB
-		if cpu < TotalCPUs || ram < gbToBytes(TotalRAM) || disk < o.OCSRequiredDisk || hostsWithDisks < o.OCSRequiredHosts { // check for master nodes requirements
-			status = o.setStatusInsufficientResources(cpu, ram, disk, hostsWithDisks, "Compact Mode")
+
+	if int64(len(hosts)) == o.config.OCSRequiredHosts { // for 3 hosts
+		TotalCPUs = o.config.OCSRequiredCompactModeCPUCount
+		TotalRAM = o.config.OCSRequiredCompactModeRAMGB
+		if !o.workerNodesRequirements(cpu, ram, disk, hostsWithDisks, TotalCPUs, TotalRAM) { // check for master nodes requirements
+			status = o.setStatusInsufficientResources(cpu, ram, disk, hostsWithDisks, compact)
 			return false, status
 		}
-		o.OCSDeploymentType = "Compact"
+		o.config.OCSDeploymentType = compact
 		status = "OCS Requirements for Compact Mode are satisfied"
 		return true, status
 	}
-	TotalCPUs = o.OCSMinimumCPUCount
-	TotalRAM = o.OCSMinimumRAMGB
-	if disk < o.OCSRequiredDisk || cpu < TotalCPUs || ram < gbToBytes(TotalRAM) || hostsWithDisks < o.OCSRequiredHosts { // check for worker nodes requirements
-		status = o.setStatusInsufficientResources(cpu, ram, disk, hostsWithDisks, "Minimal Deployment Mode")
+
+	TotalCPUs = o.config.OCSMinimumCPUCount
+	TotalRAM = o.config.OCSMinimumRAMGB
+	if !o.workerNodesRequirements(cpu, ram, disk, hostsWithDisks, TotalCPUs, TotalRAM) { // check for worker nodes requirements
+		status = o.setStatusInsufficientResources(cpu, ram, disk, hostsWithDisks, minimal)
 		return false, status
 	}
-
-	TotalCPUs = o.OCSRequiredCPUCount
-	TotalRAM = o.OCSRequiredRAMGB
+	TotalCPUs = o.config.OCSRequiredCPUCount
+	TotalRAM = o.config.OCSRequiredRAMGB
 	if cpu < TotalCPUs || ram < gbToBytes(TotalRAM) { // conditions for minimal deployment
 		status = "Requirements for OCS Minimal Deployment are satisfied"
-		o.OCSDeploymentType = "Minimal"
-		o.OCSMinimalDeployment = true
+		o.config.OCSDeploymentType = minimal
+		o.config.OCSMinimalDeployment = true
 		return true, status
 	}
 
 	status = "OCS Requirements for Standard Deployment are satisfied"
-	o.OCSDeploymentType = "Standard"
+	o.config.OCSDeploymentType = standard
 	return true, status
 }
 
-func (o *ocsValidator) setStatusInsufficientResources(cpu int64, ram int64, disk int64, hostsWithDisks int64, mode string) string {
+func (o *operator) workerNodesRequirements(cpu int64, ram int64, disk int64, hostsWithDisks int64, TotalCPUs int64, TotalRAM int64) bool {
```

Comment:
Misleading function name - the function is called for compact mode as well which means the nodes are masters. 
Other changes:
Change the fn signature to take the resources struct as an argument
Pass in the deployment mode instead of TotalRAM and TotalCPUs. Let the function determine that based on the mode. That way we can extend the function to compare more params in future without changing the signature.

It can be written like this:

func requirementSatisfied(o *operator, nodeInfo *resourceInfo, mode string) bool {
	var totalCPUs int64
	var totalRAM int64

	if mode == compact {
		totalCPUs = o.config.OCSRequiredCompactModeCPUCount
		totalRAM = o.config.OCSRequiredCompactModeRAMGB
	} else if mode == minimal {
		totalCPUs = o.config.OCSMinimumCPUCount
		totalRAM = o.config.OCSMinimumRAMGB
	} else if mode == standard {
		totalCPUs = o.config.OCSRequiredCPUCount
		totalRAM = o.config.OCSRequiredRAMGB
	}
	return nodeInfo.cpuCount >= totalCPUs && nodeInfo.ram >= gbToBytes(totalRAM) && nodeInfo.numDisks >= o.config.OCSRequiredDisk && nodeInfo.hostsWithDisks >= o.config.OCSRequiredHosts
}


And called by validate()  like this:


	if numAvailableHosts == o.config.OCSRequiredHosts { // for 3 hosts
		if !requirementSatisfied(o, nodeInfo, "compact") { // check for master nodes requirements
			status = o.setStatusInsufficientResources(nodeInfo, compact)
			return false, status
		}
		o.config.OCSDeploymentType = compact
		status = "OCS Requirements for Compact Mode are satisfied"
		return true, status
	}

	if !requirementSatisfied(o, nodeInfo, standard) { // check for worker nodes requirement
		if !requirementSatisfied(o, nodeInfo, minimal) { // check for worker nodes requirements
			status = o.setStatusInsufficientResources(nodeInfo, minimal)
			return false, status
		}
		status = "Requirements for OCS Minimal Deployment are satisfied"
		o.config.OCSDeploymentType = minimal
		o.config.OCSMinimalDeployment = true
		return true, status
	}
	status = "OCS Requirements for Standard Deployment are satisfied"
	o.config.OCSDeploymentType = standard
	return true, status
}